tokio/src/io/async_fd.rs

```diff
@@ -826,6 +826,9 @@ impl<'a, Inner: AsRawFd> AsyncFdReadyGuard<'a, Inner> {
     /// _actually observes_ that the file descriptor is _not_ ready. Do not call
     /// it simply because, for example, a read succeeded; it should be called
     /// when a read is observed to block.
+    ///
+    /// Note that that if a readiness notification occurs following the last operation
+    /// but prior to invoking `clear_ready`, it will not be cleared.
```

Comment:
I think it's really important to be very clear in the wording here, because the point we are making is tricky to understand.

Here's one suggested wording. This one tries to explain it twice with "in other words". I still don't love it though.
```suggestion
    /// This method only clears readiness events that happened before the creation of this guard. In other words, if the IO resource becomes ready between the creation of the guard and this call to `clear_ready`, then the readiness is not actually cleared.
```

Reply:
I actually like that the timing is tied to the creation of the guard. I believe it makes it clear.